silicon/backends/mhd.hh

```diff
@@ -621,5 +621,41 @@ namespace sl
   {
     return mhd_json_serve(api, std::make_tuple(), port, opts...);
   }
-  
+
+  auto file_server(const std::string& base_path)
+  {
+    static char dot = '.', slash = '/';
+    if(!base_path.empty() && base_path[base_path.size() - 1] != '/'){
+      base_path.append('/');
+    }
+    return [&base_path, &dot, slash](mhd_request* r){
```

Comment:
Prefer copies instead of refererences of base_path and dot here.
This could lead to segv if base_path get destroyed before the launch
of the server.